conjure-java-core/src/integrationInput/java/com/palantir/product/ErrorServiceAsync.java

```diff
@@ -0,0 +1,386 @@
+                    @JsonProperty("errorCode") String errorCode,
+                    @JsonProperty("errorInstanceId") String errorInstanceId,
+                    @JsonProperty("parameters") TestErrors.NotFoundParameters parameters) {
+                super(errorCode, TestErrors.NOT_FOUND.name(), errorInstanceId, parameters);
+            }
+        }
+
+        final class DifferentNamespace extends ConjureErrors.BaseEndpointError<ConjureErrors.EmptyErrorParameters>
+                implements TestMultipleErrorsAndPackagesResponse {
+            @JsonCreator
+            DifferentNamespace(
+                    @JsonProperty("errorCode") String errorCode,
+                    @JsonProperty("errorInstanceId") String errorInstanceId) {
+                super(
+                        errorCode,
+                        EndpointSpecificTwoErrors.DIFFERENT_NAMESPACE.name(),
+                        errorInstanceId,
+                        ConjureErrors.EMPTY_ERROR_PARAMETERS_INSTANCE);
+            }
+        }
+
+        final class DifferentPackage extends ConjureErrors.BaseEndpointError<ConjureErrors.EmptyErrorParameters>
+                implements TestMultipleErrorsAndPackagesResponse {
+            @JsonCreator
+            DifferentPackage(
+                    @JsonProperty("errorCode") String errorCode,
+                    @JsonProperty("errorInstanceId") String errorInstanceId) {
+                super(
+                        errorCode,
+                        com.palantir.another.EndpointSpecificErrors.DIFFERENT_PACKAGE.name(),
+                        errorInstanceId,
+                        ConjureErrors.EMPTY_ERROR_PARAMETERS_INSTANCE);
+            }
+        }
+    }
+
+    sealed interface TestEmptyBodyResponse
+            permits TestEmptyBodyResponse.Success, TestEmptyBodyResponse.InvalidArgument {
+        record Success() implements TestEmptyBodyResponse {
+            @JsonCreator
+            public static Success create() {
+                return new Success();
+            }
+        }
+
+        final class InvalidArgument extends ConjureErrors.BaseEndpointError<TestErrors.InvalidArgumentParameters>
+                implements TestEmptyBodyResponse {
+            @JsonCreator
+            InvalidArgument(
+                    @JsonProperty("errorCode") String errorCode,
+                    @JsonProperty("errorInstanceId") String errorInstanceId,
+                    @JsonProperty("parameters") TestErrors.InvalidArgumentParameters parameters) {
+                super(errorCode, TestErrors.INVALID_ARGUMENT.name(), errorInstanceId, parameters);
+            }
+        }
+    }
+
+    sealed interface TestBinaryResponse permits TestBinaryResponse.Success, TestBinaryResponse.InvalidArgument {
+        record Success(@MustBeClosed InputStream value) implements TestBinaryResponse {
+            public Success {
+                Preconditions.checkArgumentNotNull(value, "value cannot be null");
+            }
+        }
```

Comment:
When the success type is an `InputStream` or `Optional<InputStream>`, I think the base type should implement `Closeable` so that callers can use a try-with-resources block on the call without risking resource leaks.

We can bound the codegen by generating a no-op close method on the base type, then we don't need to repeat it in every error type, and can implement the real closure logic in the `Success` variant.
```java
@Override
default void close() throws IOException {}
```